pyomo/core/tests/unit/test_sos_v2.py

```diff
@@ -0,0 +1,613 @@
+# ******************************************************************************
+# ******************************************************************************
+
+import math
+import pyomo.environ as pyo
+import pyomo.common.unittest as unittest
+from pyomo.opt import check_available_solvers
+
+solver_name = "scip"
+
+solver_available = bool(check_available_solvers(solver_name))
+
+# *****************************************************************************
+# *****************************************************************************
+
+
+class TestSOS(unittest.TestCase):
+    @unittest.skipIf(not solver_available, "The solver is not available.")
```

Comment:
Another recommendation: Move this `skipIf` statement to above the class so it only has to be checked once/can be easily changed.

Reply:
Done.